packages/context/src/__tests__/acceptance/class-level-bindings.acceptance.ts

```diff
@@ -145,34 +148,181 @@ describe('Context bindings - Injecting dependencies of classes', () => {
   });
 
   it('injects a getter function', async () => {
-    ctx.bind('key').to('value');
+    ctx.bind('hash').to('123');
 
     class Store {
-      constructor(@inject.getter('key') public getter: Getter<string>) {}
+      constructor(@inject.getter('hash') public getter: Getter<string>) {}
     }
 
     ctx.bind('store').toClass(Store);
     const store = ctx.getSync<Store>('store');
 
     expect(store.getter).to.be.Function();
-    expect(await store.getter()).to.equal('value');
+    expect(await store.getter()).to.equal('123');
 
     // rebind the value to verify that getter always returns a fresh value
-    ctx.bind('key').to('new-value');
-    expect(await store.getter()).to.equal('new-value');
+    ctx.bind('hash').to('456');
+    expect(await store.getter()).to.equal('456');
+  });
+
+  describe('in SINGLETON scope', () => {
+    let requestCtx: Context;
+
+    it('throws if a getter cannot be resolved by the owning context', async () => {
+      class Store {
+        constructor(@inject.getter('hash') public getter: Getter<string>) {}
+      }
+
+      givenRequestContext(Store);
+
+      // Create the store instance from the child context
+      // Now the singleton references a `getter` to `hash` binding from
+      // the `requestCtx`
+      const store = requestCtx.getSync<Store>('store');
```

Comment:
Where is the binding key `store` coming from?

Reply:
I extract the binding key to be constants.